src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  currentTodo: Todo | null,
+  setCurrentTodoId: (id: number | null) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  currentTodo,
+  setCurrentTodoId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId).then(currentUser => {
+        setUser(currentUser);
+      });
+    }
```

Comment:
If you render modal only when you have selected todo, you don't need this check

 ```suggestion
    getUser(userId)
      .then(currentUser => {
        setUser(currentUser);
      })
      .finally(() => {
        setIsLoading(false)
      });
```